pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -486,6 +534,83 @@ func (adc *attachDetachController) nodeDelete(obj interface{}) {
 	adc.processVolumesInUse(nodeName, node.Status.VolumesInUse)
 }
 
+func (adc *attachDetachController) enqueuePVC(obj interface{}) {
+	key, err := kcache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("Couldn't get key for object %+v: %v", obj, err))
```

Comment:
Just curious: Why `runtime.HandleError`? Why not just `glog.Errorf`?

Reply:
I followed https://github.com/kubernetes/kubernetes/blob/v1.12.0-alpha.1/pkg/controller/daemon/daemon_controller.go#L316, and found [sample-controller](https://github.com/kubernetes/sample-controller/blob/master/controller.go#L341) uses `runtime.HandleError` to log errors too. Perhaps because `runtime.HandleError` is dedicated to log non-user facing ignored errors and can be extended by adding error handlers.